hermes/probe/create_file_test.go

```diff
@@ -1 +1,201 @@
+	fTwo := RandomHermesFile{13, 1000}
+	fCopy := RandomHermesFile{11, 100}
+	checksum, err := f.CheckSum() 
+	if(err!=nil){
+		t.Error(err)
+	}
+	checksumTwo, err := fTwo.CheckSum() 
+	if(err!=nil){
+		t.Error(err)
+	}
+	checksumCopy, err := fCopy.CheckSum() 
+	if(err!=nil){
+		t.Error(err)
+	}
+	if fmt.Sprintf("%x",checksum) != fmt.Sprintf("%x",checksumCopy) { //comparing checksums converted to strings as a slice can only be compared to nil
+		t.Errorf("Checksum returned different values for the same RandomHermesFiles {%v, %v}", f.ID, f.Size)
+	}
+	if fmt.Sprintf("%x",checksum) == fmt.Sprintf("%x",checksumTwo) {
+		t.Errorf("Checksum returned the same value for two different RandomHermesFiles {%v, %v} and {%v, %v}", f.ID, f.Size, fTwo.ID, fTwo.Size)
+	}
+
+}
+
+type fakeBucket struct {
+	attrs   *storage.BucketAttrs
+	objects map[string][]byte
+}
+
+type fakeClient struct {
+	stiface.Client
+	buckets map[string]*fakeBucket
+}
+
+type fakeBucketHandle struct {
+	stiface.BucketHandle
+	c    *fakeClient
+	name string
+}
+
+type fakeObjectHandle struct {
+	stiface.ObjectHandle
+	c          *fakeClient
+	bucketName string
+	name       string
+}
+
+type fakeWriter struct {
+	stiface.Writer
+	obj fakeObjectHandle
+	buf bytes.Buffer
+}
+
+func newFakeClient() stiface.Client {
+	return &fakeClient{buckets: map[string]*fakeBucket{}}
+}
+
+func (o fakeObjectHandle) NewWriter(context.Context) stiface.Writer {
+	return &fakeWriter{obj: o}
+}
+
+func (b fakeBucketHandle) Create(_ context.Context, _ string, attrs *storage.BucketAttrs) error {
+	if _, ok := b.c.buckets[b.name]; ok {
+		return fmt.Errorf("bucket %q already exists", b.name)
+	}
+	if attrs == nil {
+		attrs = &storage.BucketAttrs{}
+	}
+	attrs.Name = b.name
+	b.c.buckets[b.name] = &fakeBucket{attrs: attrs, objects: map[string][]byte{}}
+	return nil
+}
+
+func (b fakeBucketHandle) Object(name string) stiface.ObjectHandle {
+	return fakeObjectHandle{c: b.c, bucketName: b.name, name: name}
+}
+
+func (c *fakeClient) Bucket(name string) stiface.BucketHandle {
+	return fakeBucketHandle{c: c, name: name}
+}
+
+func (w *fakeWriter) Write(data []byte) (int, error) {
+	return w.buf.Write(data)
+}
+
+func (b fakeBucketHandle) Attrs(context.Context) (*storage.BucketAttrs, error) {
+	bkt, ok := b.c.buckets[b.name]
+	if !ok {
+		return nil, fmt.Errorf("bucket %q does not exist", b.name)
+	}
+	return bkt.attrs, nil
+}
+
+func (w *fakeWriter) Close() error {
+	bkt, ok := w.obj.c.buckets[w.obj.bucketName]
+	if !ok {
+		return fmt.Errorf("bucket %q not found", w.obj.bucketName)
+	}
+	bkt.objects[w.obj.name] = w.buf.Bytes()
+	return nil
+}
+
+func TestCreateFile(t *testing.T){
+	ctx := context.Background()
+	bucketName := "test_bucket_probe0"
+	client := newFakeClient()
+	fbh := client.Bucket(bucketName) // fakeBucketHandle
+	err := fbh.Create(ctx, bucketName, nil) // creates the bucket with name "test_bucket_probe0"
+	mp := &MonitorProbe{}
+	var fileID int64 = 6
+	fileSize := 50
+	target := "ossdawdler-step2020"
```

Comment:
Does this have to be the same as a GCP project? If not, we can choose something more generic instead for test purposes.

Reply:
tagret string has now been removed